ui/lib/pki/addon/routes/certificates/index.js

```diff
@@ -1,16 +1,25 @@
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
 
-export default class PkiCertificatesIndexRoute extends Route {
+export default class RolesIndexRoute extends Route {
   @service store;
   @service secretMountPath;
+  @service pathHelp;
+
+  beforeModel() {
+    // Must call this promise before the model hook otherwise it doesn't add OpenApi to record.
+    return this.pathHelp.getNewModel('pki/pki-certificate-engine', 'pki');
+  }
 
   model() {
     return this.store
       .query('pki/pki-certificate-engine', { backend: this.secretMountPath.currentPath })
+      .then((certificateModel) => {
+        return { certificateModel, parentModel: this.modelFor('certificates') };
+      })
       .catch((err) => {
         if (err.httpStatus === 404) {
-          return [];
+          return { parentModel: this.modelFor('certificates') };
```

Comment:
needed for the EmptyState. The issuers.index or roles.index, etc routes return a model that does not have the information needed for certain links. That's why we have to fetch the parentModel (a secret-engine model) which returns information we need in the EmptyState and SecretListHeader, which will show even if we get a 404 on the LIST request. (The empty states are conditional. The language and links change if the pki engine has been mounted and that part of the work has not been done yet).